fearless/Modules/Wallet/TransferConfirm/TransferConfirmViewModelFactory.swift

```diff
@@ -108,20 +67,56 @@ final class TransferConfirmViewModelFactory {
         payload: ConfirmationPayload,
         locale: Locale
     ) {
-        guard let asset = assets.first(where: { $0.identifier == payload.transferInfo.asset }) else {
+        guard let asset = assets
+            .first(where: { $0.identifier == payload.transferInfo.asset }),
+            let assetId = WalletAssetId(rawValue: asset.identifier)
+        else {
             return
         }
 
-        let formatter = amountFormatterFactory.createInputFormatter(for: asset)
+        let formatter = amountFormatterFactory.createInputFormatter(for: asset).value(for: locale)
+        let balanceFormatter = amountFormatterFactory.createTokenFormatter(for: asset).value(for: locale)
 
         let decimalAmount = payload.transferInfo.amount.decimalValue
 
-        guard let amount = formatter.value(for: locale).string(from: decimalAmount as NSNumber) else {
+        guard let amount = formatter.string(from: decimalAmount as NSNumber) else {
```

Comment:
Also here we can extract formatted amount from balanceViewModelFactory together with price